# PLOTLY KÜTÜPHANESİ

2012 yılında kurulmuş bir kütüphanedir.

Plotly, Veri görselleştirme ve dashboard oluşturmaya yarayan çeşitli fonksiyonlar içeren bir kütüphanedir. Bu fonksiyonları kullanarak klasik grafikler yerine interaktif grafikler elde edebiliriz.

Plotly'ın kullanıldığı yerler:

🟡 Python

🟢 R & Shiny

🟣 Matlab

🔵 Javascript

vs. bir çok yerde kullanımı söz konusu.

Plotly kütüphanesinde, Seaborn kütüphanesinden farklı olarak grafik üzerinde gezinirken bulunduğumuz konuma karşılık gelen değeri görebiliyoruz.

<font color="Blue">
Üniversitelerin dünya sıralamalarını ifade eden bir veri seti kullanacağız.
<font color="Blue"><br>    
timesData.csv isimli veri seti içeriğinde bulunan sütunlardan bahsedelim.
<font color="Blue"><br>
world_rank: Dünya sıralaması
<font color="Blue"><br>
university_name: Üniversite ismi
<font color="Blue"><br>
country: Üniversitenin ait olduğu kıta
<font color="Blue"><br>
teaching: Üniversitenin öğretme değeri
<font color="Blue"><br>
international: Uluslararası puanı
<font color="Blue"><br>
research: Araştırma puanı
<font color="Blue"><br>
citations: Alıntı yapma puanı
<font color="Blue"><br>
income: Gelir bilgisi
<font color="Blue"><br>
total_score: Toplam skoru
<font color="Blue"><br>
num_students: Kaç tane öğrenciye sahip olduğu bilgisi
<font color="Blue"><br>
student_staff_ratio: Öğrenci için çalışan oranı
<font color="Blue"><br>
international_students: Uluslararası öğrencilerin oranı
<font color="Blue"><br>
female_male_ratio: Erkek kadın oranı
<font color="Blue"><br>
year: Değerlendirmenin yani sıralamanın yapıldığı yıl bilgisi

Plotly kütüphanesini kullanarak gerçekleştirebileceğimiz plot'lar şunlardır:

<font color="Green"><br>
line plots, scatter plots, area charts, bar charts, error bars, box plots, histograms, heatmaps, subplots, multiple-axes, polar charts ve bubble charts.


Kütüphaneleri import ederek işe başlayalım.

In [1]:
pip install chart_studio

Note: you may need to restart the kernel to use updated packages.


In [2]:
pip install plotly --upgrade

Note: you may need to restart the kernel to use updated packages.


In [3]:
conda install -c conda-forge wordcloud

Solving environment: ...working... done

# All requested packages already installed.


Note: you may need to restart the kernel to use updated packages.


In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#plotly
#import plotly.plotly as py
#from plotly.offline import init_notebook_mode, iplot 
#init_notebook_mode(connected=True)
import chart_studio.plotly as py
import plotly.graph_objs as go

#word cloud library: Word cloud pyplot kütüphanesine ait değil fakat güzel bir plot olmasından dolayı buda anlatılacaktır.
from wordcloud import WordCloud

Veri setini çağıralım.

In [5]:
timesData = pd.read_csv("timesData.csv")

Şimdi de veri setini inceleyelim.

In [6]:
timesData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2603 entries, 0 to 2602
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   world_rank              2603 non-null   object 
 1   university_name         2603 non-null   object 
 2   country                 2603 non-null   object 
 3   teaching                2603 non-null   float64
 4   international           2603 non-null   object 
 5   research                2603 non-null   float64
 6   citations               2603 non-null   float64
 7   income                  2603 non-null   object 
 8   total_score             2603 non-null   object 
 9   num_students            2544 non-null   object 
 10  student_staff_ratio     2544 non-null   float64
 11  international_students  2536 non-null   object 
 12  female_male_ratio       2370 non-null   object 
 13  year                    2603 non-null   int64  
dtypes: float64(4), int64(1), object(9)
memor

Bir DataFrame olduğunu ve veri seti içerisinde 2603 tane satır olduğunu görüyoruz. 14 tane feature vardır ve feature'leri görüyoruz. Sütunların 4 tanesi float tipinde, 1 tanesi int tipinde ve 9 tanesi string tipindedir. Toplam hafıza kullanımı 284.8 KB'dir.

Veri seti içeriğine bir göz atalım.

In [7]:
timesData.head(10)

world_rank                        university_name                   country  \
0          1                     Harvard University  United States of America   
1          2     California Institute of Technology  United States of America   
2          3  Massachusetts Institute of Technology  United States of America   
3          4                    Stanford University  United States of America   
4          5                   Princeton University  United States of America   
5          6                University of Cambridge            United Kingdom   
6          6                   University of Oxford            United Kingdom   
7          8     University of California, Berkeley  United States of America   
8          9                Imperial College London            United Kingdom   
9         10                        Yale University  United States of America   

   teaching international  research  citations income total_score  \
0      99.7          72.4      98.7       98.8   34.5        96.1   
1      97.7          54.6      98.0       99.9   83.7        96.0   
2      97.8          82.3      91.4       99.9   87.5        95.6   
3      98.3          29.5      98.1       99.2   64.3        94.3   
4      90.9          70.3      95.4       99.9      -        94.2   
5      90.5          77.7      94.1       94.0   57.0        91.2   
6      88.2          77.2      93.9       95.1   73.5        91.2   
7      84.2          39.6      99.3       97.8      -        91.1   
8      89.2          90.0      94.5       88.3   92.9        90.6   
9      92.1          59.2      89.7       91.5      -        89.5   

  num_students  student_staff_ratio international_students female_male_ratio  \
0       20,152                  8.9                    25%               NaN   
1        2,243                  6.9                    27%           33 : 67   
2       11,074                  9.0                    33%           37 : 63   
3       15,596                  7.8                    22%           42 : 58   
4        7,929                  8.4                    27%           45 : 55   
5       18,812                 11.8                    34%           46 : 54   
6       19,919                 11.6                    34%           46 : 54   
7       36,186                 16.4                    15%           50 : 50   
8       15,060                 11.7                    51%           37 : 63   
9       11,751                  4.4                    20%           50 : 50   

   year  
0  2011  
1  2011  
2  2011  
3  2011  
4  2011  
5  2011  
6  2011  
7  2011  
8  2011  
9  2011

Çıktıyı yorumlayalım. 2011 yılında yapılan sıralamada Hardvard University 1.sırada yer almakta ve bu değerlendirmede kadın erkek oranı NaN yani kadın erkek oranına dair bir hesap alınmamış.